client.go

```diff
@@ -21,6 +21,7 @@ type Client struct {
 	Auth     Auth
 	Logger   logr.Logger
 	Registry *registrar.Registry
+	metadata *bmc.Metadata
```

Comment:
This approach is cleaner, and more user friendly than the previous, 
Although, since `Client.metadata` isn't under any mutex, the caller needs to be careful with interpreting the data held. I don't think this is a blocker to merge though, and given this object is now available it can be changed internally without breaking the API.

Reply:
I can update with a mutex on writes. I only think this matters if users are making calls to multiple *Client.X functions/methods across goroutines.